packages/firestore/test/unit/remote/bloom_filter.test.ts

```diff
@@ -0,0 +1,219 @@
+      'Invalid hash count: -1'
+    );
+  });
+
+  it('should throw error if hash count is 0 for non empty bloom filter', () => {
+    expect(() => new BloomFilter(new Uint8Array(1), 1, 0)).to.throw(
+      'Invalid hash count: 0'
+    );
+  });
+
+  it('should be able to process non standard characters', () => {
+    // A non-empty BloomFilter object with 1 insertion : "ÀÒ∑"
+    const bloomFilter = new BloomFilter(new Uint8Array([237, 5]), 5, 8);
+    expect(bloomFilter.mightContain('ÀÒ∑')).to.be.true;
+    expect(bloomFilter.mightContain('Ò∑À')).to.be.false;
+  });
+
+  it('mightContain in empty bloom filter should always return false', () => {
+    const bloomFilter = new BloomFilter(new Uint8Array(0), 0, 0);
+    expect(bloomFilter.mightContain('abc')).to.be.false;
+    expect(bloomFilter.mightContain('def')).to.be.false;
+  });
+
+  it('mightContain should always return false for empty string', () => {
+    const emptyBloomFilter = new BloomFilter(new Uint8Array(0), 0, 0);
+    const nonEmptyBloomFilter = new BloomFilter(
+      new Uint8Array([255, 255, 255]),
+      1,
+      16
+    );
+    expect(emptyBloomFilter.mightContain('')).to.be.false;
+    expect(nonEmptyBloomFilter.mightContain('')).to.be.false;
+  });
+
+  /**
+   * Golden tests are generated by backend based on inserting n number of
+   * document paths into a bloom filter.
+   *
+   * Full document path is generated by concatenating documentPrefix and number
+   * n, eg, projects/project-1/databases/database-1/documents/coll/doc12.
+   *
+   * The test result is generated by checking the membership of documents from
+   * documentPrefix+0 to documentPrefix+2n. The membership results from 0 to n
+   * is expected to be true, and the membership results from n to 2n is
+   * expected to be false with some false positive results.
+   */
+  describe('BloomFilter golden tests', () => {
+    const documentPrefix =
+      'projects/project-1/databases/database-1/documents/coll/doc';
+
+    interface TestDataType {
+      bits: {
+        bitmap: string;
+        padding: number;
+      };
+      hashCount: number;
+    }
+
+    interface TestResultType {
+      membershipTestResults: string;
+    }
+
+    function decodeBase64ToUint8Array(encoded: string): Uint8Array {
+      return ByteString.fromBase64String(encoded).toUint8Array();
+    }
+
+    function testBloomFilterAgainstExpectedResult(
+      bloomFilterInputs: TestDataType,
+      expectedResult: TestResultType
+    ): void {
+      const {
+        bits: { bitmap, padding },
+        hashCount
+      } = bloomFilterInputs;
+      const { membershipTestResults } = expectedResult;
+
+      const bloomFilter = new BloomFilter(
+        decodeBase64ToUint8Array(bitmap),
+        padding,
+        hashCount
+      );
+      for (let i = 0; i < membershipTestResults.length; i++) {
+        const expectedMembershipResult =
+          membershipTestResults[i] === '1' ? true : false;
```

Comment:
This ternary statement can be simplified to simply `membershipTestResults[i] === '1'`.